instrumentation/aws-sdk/aws-sdk-1.11/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/awssdk/v1_11/S3ClientTest.java

```diff
@@ -0,0 +1,155 @@
+import static io.opentelemetry.semconv.incubating.RpcIncubatingAttributes.RPC_SERVICE;
+import static io.opentelemetry.semconv.incubating.RpcIncubatingAttributes.RPC_SYSTEM;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.Request;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.NoOpSigner;
+import com.amazonaws.auth.SignerFactory;
+import com.amazonaws.handlers.RequestHandler2;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.instrumentation.awssdk.v1_11.AbstractS3ClientTest;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class S3ClientTest extends AbstractS3ClientTest {
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Override
+  protected InstrumentationExtension testing() {
+    return testing;
+  }
+
+  @Override
+  public AmazonS3ClientBuilder configureClient(AmazonS3ClientBuilder client) {
+    return client;
+  }
+
+  // Verify agent instruments old and new construction patterns.
+  @ParameterizedTest
+  @MethodSource("provideS3Arguments")
+  void testRequestHandlerIsHookedUpWithBuilder(boolean addHandler, int size, int position)
+      throws Exception {
+    AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard().withRegion(Regions.US_EAST_1);
+
+    if (addHandler) {
+      builder.withRequestHandlers(new RequestHandler2() {});
+    }
+    AmazonS3 client = builder.build();
+
+    List<RequestHandler2> requestHandler2s = extractRequestHandlers(client);
+    assertThat(requestHandler2s).isNotNull();
+    assertThat(requestHandler2s.size()).isEqualTo(size);
+    assertThat(requestHandler2s.get(position).getClass().getSimpleName())
+        .isEqualTo("TracingRequestHandler");
+  }
+
+  private static Stream<Arguments> provideS3Arguments() {
+    return Stream.of(Arguments.of(true, 2, 1), Arguments.of(false, 1, 0));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideS3Arguments")
+  @SuppressWarnings("deprecation") // AmazonS3Client constructor is deprecated
+  void testRequestHandlerIsHookedUpWithConstructor(boolean addHandler, int size) throws Exception {
+    String accessKey = "asdf";
```

Comment:
The following variable definitions are unnecessary, since they are just used once.